src/compile-middleware.ts

```diff
@@ -115,17 +115,28 @@ export function babelCompile(
         transformModules: forceCompile || !capabilities.has('modules'),
       };
       const optionsLeader = JSON.stringify(options);
-      const cached = babelCompileCache.get(optionsLeader + body);
+      const cached = babelCompileCache.get(optionsLeader + request.path + body);
       if (cached !== undefined) {
         return cached;
       }
 
       let transformed;
       const contentType = getContentType(response);
+      let filePath = path.resolve(path.join(rootdir, request.path));
+      // The file path needs to include the filename for correct relative path
+      // calculation
+      const stat = fs.statSync(filePath);
```

Comment:
I guess we should avoid sync calls since we could be handling multiple requests? use fs-extra?

Reply:
`transform()` is sync, unfortunately. It really should be async.